redis-lock/src/main/java/com/netflix/conductor/locking/redis/RedisLock.java

```diff
@@ -0,0 +1,129 @@
+        RedisLockConfiguration.REDIS_SERVER_TYPE redisServerType = configuration.getRedisServerType();
+        try {
+            redisServerType = configuration.getRedisServerType();
+        } catch (IllegalArgumentException ie) {
+            final String message = "Invalid Redis server type: " + configuration.getRedisServerType()
+                    + ", supported values are: " + Arrays.toString(configuration.getRedisServerType().values());
+            LOGGER.error(message);
+            throw new ProvisionException(message, ie);
+        }
+        String redisServerAddress = configuration.getRedisServerAddress();
+
+        redisConfig = new Config();
+
+        switch (redisServerType) {
+            case SINGLE:
+                redisConfig.useSingleServer().setAddress(redisServerAddress).setTimeout(connectionTimeout);
+                break;
+            case CLUSTER:
+                redisConfig.useClusterServers()
+                        .setScanInterval(2000) // cluster state scan interval in milliseconds
+                        .addNodeAddress(redisServerAddress.split(","))
+                        .setTimeout(connectionTimeout);
+                break;
+            case SENTINEL:
+                redisConfig.useSentinelServers().addSentinelAddress(redisServerAddress).setTimeout(connectionTimeout);
+                break;
+        }
+
+        redisson = Redisson.create(redisConfig);
+    }
+
+    @Override
+    public void acquireLock(String lockId) {
+        RLock lock = redisson.getLock(parseLockId(lockId));
+        lock.lock();
+    }
+
+    @Override
+    public boolean acquireLock(String lockId, long timeToTry, TimeUnit unit) {
+        RLock lock = redisson.getLock(parseLockId(lockId));
+        try {
+            return lock.tryLock(timeToTry, unit);
+        } catch (Exception e) {
+            return handleAcquireLockFailure(lockId, e);
+        }
+    }
+
+    /**
+     *
+     * @param lockId resource to lock on
+     * @param timeToTry blocks up to timeToTry duration in attempt to acquire the lock
+     * @param leaseTime Lock lease expiration duration. Redisson default is -1, meaning it holds the lock until explicitly unlocked.
+     * @param unit time unit
+     * @return
+     */
+    @Override
+    public boolean acquireLock(String lockId, long timeToTry, long leaseTime, TimeUnit unit) {
+        RLock lock = redisson.getLock(parseLockId(lockId));
+        try {
+            return lock.tryLock(timeToTry, leaseTime, unit);
+        } catch (Exception e) {
+            return handleAcquireLockFailure(lockId, e);
+        }
+    }
+
+    @Override
+    public void releaseLock(String lockId) {
+        RLock lock = redisson.getLock(parseLockId(lockId));
+        try {
+            lock.unlock();
+        } catch (IllegalMonitorStateException e) {
+            // Releasing a lock twice using Redisson can cause this exception, which can be ignored.
```

Comment:
Do we need to log this even if it is being ignored?

Reply:
Being a re-entrant lock, We'd call release lock several times from the same thread, and this adds unnecessary noise to the logs.